content/en/profiler/profiler_troubleshooting.md

```diff
@@ -233,6 +233,29 @@ Datadog.configure do |c|
 end
 ```
 
+## Unexpected run-time failures and errors from Ruby gems that use native extensions in `dd-trace-rb` 1.11.0+
+
+Starting from `dd-trace-rb` 1.11.0, the "CPU Profiling 2.0" profiler gathers data by sending `SIGPROF` unix signals to Ruby applications, enabling finer-grained data gathering.
+
+Sending `SIGPROF` is a common profiling approach, and may cause system calls from native extensions/libraries to be interrupted [with a system `EINTR` error code](https://man7.org/linux/man-pages/man7/signal.7.html#:~:text=Interruption%20of%20system%20calls%20and%20library%20functions%20by%20signal%20handlers).
+Rarely, native extensions or libraries called by them may be have missing or incorrect error handling for the `EINTR` error code.
+
+One known instance of this issue is when using the `mysql2` gem together with versions of libmysqlclient [older than 8.0.0](https://bugs.mysql.com/bug.php?id=83109). The affected libmysqlclient version is known to be present on Ubuntu 18.04, but not 20.04 and later releases.
+**Note**: For this case, the profiler auto-detects when the `mysql2` gem is in use and auto-applies the solution described below.
```

Comment:
I'm a bit less sure about this suggestion. My intention is that the whole paragraph is supposed to sound like "We actually found an example of this issue... and we auto-detect this one, so don't worry". (And later on we ask customers to report these issues to us, so that we can extend this auto-detection)

I initially added this as a "Note that for this case...". The linter told me to add a `**Note**` instead, which already is kinda weird because it makes that part pop up in a weird way from the paragraph, but I went with it.

But completely separating the note from the paragraph I think breaks the connection I originally intended, and seems to make it an independent note. But then the note starts with "For this case", which is supposed to refer to the previous sentence.

So I think if we really go with the separation, we need to rewrite the "for this case as well". Any suggestions?

Reply:
Got it. That makes sense. I'll update my suggestion so that it remains one paragraph.